instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/DispatcherServletInstrumentation.java

```diff
@@ -84,32 +77,24 @@ public static void onEnter(
         @Advice.Argument(0) ModelAndView mv,
         @Advice.Local("otelContext") Context context,
         @Advice.Local("otelScope") Scope scope) {
-      context = tracer().startSpan(mv);
-      scope = context.makeCurrent();
+      Context parentContext = currentContext();
+      if (modelAndViewInstrumenter().shouldStart(parentContext, mv)) {
+        context = modelAndViewInstrumenter().start(parentContext, mv);
+        scope = context.makeCurrent();
+      }
     }
 
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void stopSpan(
+        @Advice.Argument(0) ModelAndView mv,
         @Advice.Thrown Throwable throwable,
         @Advice.Local("otelContext") Context context,
         @Advice.Local("otelScope") Scope scope) {
-      scope.close();
-      if (throwable == null) {
-        tracer().end(context);
-      } else {
-        tracer().endExceptionally(context, throwable);
-      }
-    }
-  }
-
-  public static class ErrorHandlerAdvice {
-    @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void nameResource(@Advice.Argument(3) Exception exception) {
-      if (exception != null) {
-        // It is fine to set status=ERROR here, end(Context, Response) call will overwrite it if the
-        // exception turns out to be a valid result
-        tracer().onException(Java8BytecodeBridge.currentContext(), exception);
```

Comment:
this didn't fit the Instrumenters well, but doesn't seem needed after removing `hasExceptionOnServerSpan=true` from the test